lib/config/config-loader.js

```diff
@@ -512,9 +407,21 @@ class ConfigLoader {
             throw new Error(`Could not find config file for ${fileOrDirPath}`);
         }
 
-        const { configFilePath } = this.#configFilePaths.get(absoluteDirPath);
+        const configFilePathInfo = this.#configFilePaths.get(absoluteDirPath);
+
+        if (typeof configFilePathInfo.then === "function") {
+            throw new Error(`Config file path for ${fileOrDirPath} has not yet been calculated`);
```

Comment:
Is's assumed that `configFilePathInfo` will be pending (and not rejected) here, based on the error message?

Reply:
Depending on when this is called, if this is a promise, it can be in any state: pending, fulfilled but not processed yet, or rejected. I think the message covers the first two cases. Maybe we should add more text to cover the third case as well?

Reply:
I'm not sure if the `configFilePathInfo` can be fulfilled, because the promise in the `#configFilePaths` map is synchronously replaced with its awaited value when it fulfills before any other handlers run. But it could be rejected or pending I think if `ESLint#getRulesMetaForResults` is called improperly. It's unlikely that a user will see this error message, but feel free to extend it if you want.

Reply:
I updated error messages in https://github.com/eslint/eslint/pull/19042/commits/8b1481e10aa924b95f9735d14da16e6e73c69ac6